pkg/kine/drivers/generic/generic.go

```diff
@@ -471,8 +511,17 @@ func (d *Generic) Count(ctx context.Context, prefix string) (int64, int64, error
 
 func (d *Generic) CurrentRevision(ctx context.Context) (int64, error) {
 	var id int64
+	var err error
+
+	done, err := d.ac.Admit(ctx)
+	if err != nil {
+		err = fmt.Errorf("denied: %w", err)
+		return 0, err
```

Comment:
also in other places

```suggestion
		return 0, fmt.Errorf("denied: %w", err)
```